### PR TITLE
Ma/planetscale

### PR DESCRIPTION
**Pull request by markusahlstrand**
After having issues with the new version of D1 I migrated over to planetscale. Also consolidated the migrations.

### CODE REVIEW

File: src/migrate/index.ts [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
 import migrations from "./migrations";
 import ReferenceMigrationProvider from "./ReferenceMigrationProvider";
+import { getDb } from '../services/db';
 
 export async function migrateToLatest(ctx: Context<Env>) {
   const provider = new ReferenceMigrationProvider(migrations);
-  const db = new Kysely<any>({
-    dialect: new D1Dialect({ database: ctx.env.AUTH_DB }),
-  });
+  const db = getDb(ctx.env);
```
Comment (markusahlstrand) [inline src/migrate/index.ts:11]: Moved to use the db service instead of defining twice

File: src/migrate/migrations/2022-12-11T09:17:35_init.ts [hunk 2]
```diff
@@ -1,100 +1,105 @@
...
   await db.schema
     .createTable("tenants")
-    .addColumn("id", "varchar", (col) => col.notNull().primaryKey())
-    .addColumn("name", "varchar")
-    .addColumn("audience", "varchar")
-    .addColumn("sender_email", "varchar")
-    .addColumn("sender_name", "varchar")
-    .addColumn("created_at", "varchar")
-    .addColumn("modified_at", "varchar")
+    .addColumn("id", "varchar(255)", (col) => col.primaryKey())
```
Comment (markusahlstrand) [inline src/migrate/migrations/2022-12-11T09:17:35_init.ts:7]: Planetscale wants to know the length..

File: src/services/db.ts [hunk 3]
```diff
@@ -1,24 +1,29 @@
...
 import { Env } from "../types/Env";
+import { PlanetScaleDialect } from "kysely-planetscale";
 
 let _db: Kysely<Database>;
 
 export function getDb(env: Env): Kysely<Database> {
   if (!_db) {
-    _db = new Kysely<Database>({
-      dialect: new D1Dialect({ database: env.AUTH_DB }),
+    // _db = new Kysely<Database>({
```
Comment (markusahlstrand) [inline src/services/db.ts:14]: Keeping this for now as I hope we can move back.

File: jest.setup.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+// Polyfill for encoding which isn't present globally in jsdom
+import { TextEncoder, TextDecoder } from "util";
```
Comment (markusahlstrand) [inline jest.setup.js:2]: This adapter uses TextEncoder which for some reason isn't part of jsdom.. odd.

File: src/handlers/update-user.ts [hunk 5]
```diff
@@ -33,9 +33,16 @@ export async function updateUser(env: Env, tenantId: string, email: string) {
...
-  await db
-    .insertInto("users")
-    .values(user)
-    .onConflict((oc) => oc.columns(["id", "tenantId"]).doUpdateSet(user))
-    .execute();
+  try {
+    await db
+      .insertInto("users")
+      .values(user)
+      // .onConflict((oc) => oc.columns(["id", "tenantId"]).doUpdateSet(user))
```
Comment (markusahlstrand) [inline src/handlers/update-user.ts:40]: The planetscale kysely adapter doesn't support this really slick onConflict function. Have to brute force it a bit..

File: package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "cloudworker-auth",
-  "version": "1.15.5",
+  "name": "auth2",
```
Comment (danielgent) [inline package.json:2]: Should we rename the project here on Github? :see_no_evil: 

GitHub sets redirects so we don't need to change anything anywhere after

File: package.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
...
+  "version": "1.15.3",
```
Comment (danielgent) [inline package.json:3]: Cool it's a negative release! :smile: 

image

File: package.json [hunk 8]
```diff
@@ -43,6 +43,7 @@
     ]
   },
   "dependencies": {
+    "@planetscale/database": "^1.10.0",
```
Comment (danielgent) [inline package.json:46]: TIL! They must have been happy with that name :joy:

File: src/types/Profile.ts [hunk 9]
```diff
@@ -73,3 +73,5 @@ export const ProfileSchema = z.object({
   locale: z.string().optional(),
   connections: z.array(ConnectionSchema),
 });
+
+// export type Profile = {} & z.infer<typeof ProfileSchema>;
```
Comment (danielgent) [inline src/types/Profile.ts:77]:
```suggestion
```

No reason to leave this commented? :smile:

File: src/routes/tsoa/applications.ts [hunk 10]
```diff
@@ -197,11 +197,22 @@ export class ApplicationsController extends Controller {
...
-    await db
-      .insertInto("applications")
-      .values(application)
-      .onConflict((oc) => oc.column("id").doUpdateSet(body))
-      .execute();
+    try {
+      await db
+        .insertInto("applications")
+        .values(application)
+        // .onConflict((oc) => oc.column("id").doUpdateSet(body))
```
Comment (danielgent) [inline src/routes/tsoa/applications.ts:204]: are you planning to revert this back to just this line at some point? If so, worth adding a comment? `planetscale currently not fully supported with kysely`

**Review by danielgent — APPROVED** [review]
Some questions :smile:

File: src/routes/tsoa/applications.ts (hunk 10 above)
Comment (markusahlstrand) [reply src/routes/tsoa/applications.ts:204]: Yeah.. kind of hoping we can revert this at some point..

File: package.json (hunk 6 above)
Comment (markusahlstrand) [reply package.json:2]: Yepp, I think auth2 is the official name by now :)

**Pull request merged**